### ISSUE TITLE
Provide better feedback when a file cannot be loaded

### ISSUE THREAD

**Issue by r03ert0** [neuroanatomy/BrainBox#266]
There are common reasons for failure, for example, trying to use private data from google, or a file that's too large (and google will not serve), etc.

**Comment by adityaofficial10** [comment]
Hey @r03ert0 @katjaq !
Can I work on this issue?

**Comment by katjaq** [comment]
Hello @adityaofficial10 ! How nice that you would love to work on that issue! :D 

I will write some info here, and we just thought it could be nice to point people to an FAQ page (as part of the error feedback?) 
I will paste link to that sort of wiki here in short. (We'll use github wiki :) 

so here some info for you that should help implement better feedback:
* if it is a file from google drive:
    * the file needs to be smaller than 30MB, otherwise google will block download
    * the link obtained from g-drive needs to be modified :   
        replace "file/d/" by "uc?id=" and delete "/view?usp=sharing" at the end of the url  
        You can check that if you paste your url in a private window, you should be able to download the file  

* if it is a dropbox link, users need to take the original share link and 
    * replace "www" by "dl" and 
    * delete everything after "nii.gz"
    * You can check that if you paste your url in a private window, you should be able to download the file, then it means it should work in BrainBox :)
    
and so on, we will add more detailed info on more sources in the FAQ

Thank you so much for your help!! :)

**Comment by katjaq** [comment]
We filled some info in the wiki https://github.com/neuroanatomy/BrainBox/wiki/Frequently-Asked-Questions
Hope this helps ! :D Thank so much!

**Comment by adityaofficial10** [comment]
@katjaq 
Sorry for the late reply..
Thanks a lot for your help and the resources.
So I think we can provide a link to the wiki and also giving out some feedback like 'Check out the FAQs to find out the possible reasons.', etc.
Does this look good?

**Comment by r03ert0** [comment]
thank you for your help @adityaofficial10 !
maybe you can try a couple of things:

- try to open in brainbox a file that doesn't exist
- try to open a file from gdrive that is not publicly available
- try to open a file from grdive that's >30MB.

These should all lead to errors. The aim will be to catch those errors and report them in a way that could be useful for the different cases.

Finally, I don't know what would be the best way of reporting the errors because there are 2 different places where they can emerge. First, if you enter the URL of your file in the home page, that will open the brainbox.pasteur.fr/mri/ route. Second, if your URL is part of a project, then the error-triggering file could be opened through the brainbox.pasteur.fr/project/ route. Ideally, we would like to have the code giving feedback on the error in a single place, and not repeated for the /mri and /project routes.

**Comment by adityaofficial10** [comment]
Thanks a lot for this @r03ert0 .
I'll quickly devise a plan and comment here.
We can move forward with that..

**Comment by r03ert0** [comment]
super cool!
Actually, we just saw a case of a person that was entering the url for a directory (https://thing.com/directory/) instead of pointing to a file, which made BrainBox crash without any grace :-/

**Comment by adityaofficial10** [comment]
Hey @r03ert0 @katjaq !
I have attached a screenshot.
<img width="1440" alt="Screenshot 2021-04-27 at 10 37 44 PM" src="https://user-images.githubusercontent.com/65034276/116283232-44c4ec80-a7a9-11eb-8cc5-6ebf35b95dcf.png">
Is this the screen when file upload fails?

**Comment by r03ert0** [comment]
yup... at some point there was an error message, but it's not showing up anymore. You should get an error message displayed in the Web console.
It shouldn't be hard to add a test to check that the error message actually displays (to prevent it from disappearing again in the future...)

**Comment by adityaofficial10** [comment]
@r03ert0 
Yeah sure.
I have examined the message in the console.
I'll try to compile a good feedback and come back shortly.

**Comment by adityaofficial10** [comment]
@r03ert0 @katjaq 
The plan is to create a new category of routes namely ' error '.
This will be useful in the future as well if we discover some other errors.
So I've added 'error/loadingFile' currently.
We'll collect the error from the route and redirect the response to the error route.
And we can then render the error feedback page using that route.
Does it look good?

**Comment by katjaq** [comment]
Hey @adityaofficial10 – sounds great!! 👍  Green light from @r03ert0 and me to go ahead! 🎉 Thank you 😊

**Comment by adityaofficial10** [comment]
@r03ert0 @katjaq 
It seems the issue is far deeper.
Actually I tried many things but there is something which stops the route to render the error page.
Something like this:
<img width="665" alt="Screenshot 2021-05-05 at 6 34 07 PM" src="https://user-images.githubusercontent.com/65034276/117145256-95070480-add0-11eb-9cf2-45cef27599ab.png">
First I need to examine in detail about what happens here.
So I might need some time for this.
Thanks & Regards.

**Comment by adityaofficial10** [comment]
Hey @katjaq !
Can you please create a project board for this repo?
I am not an internal member.😅
I have decided to start with basic cleaning up tasks like fixing small bugs and completing todo tasks.
I am still working on the error feedback and hopefully I'll crack it in some days.
Regards & Thanks again..

**Comment by katjaq** [comment]
Yeyy! thanks for asking! I created a project – but let me know how i could make it better (eg smaller cards instead of one big one?)
also, are you able to add tasks yourself ? like you mentioned these little bugs you found along the way, that you may want to add there?

**Comment by adityaofficial10** [comment]
@katjaq 
Thanks a lot for that.
Its pretty great and I'd say the cards are pretty good to start with.
Unfortunately, I can't add tasks there.
Maybe you can try assigning me a card there and we'd see then.
Regards..

**Comment by r03ert0** [comment]
You can add issues, and then we can put them as cards : )

**Comment by adityaofficial10** [comment]
> You can add issues, and then we can put them as cards : )

Sure, we'd work this way.
Thanks & Regards...
